### ISSUE TITLE
Incorrect type inference for unannotated parameters in `dash` library

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#4030]
When pyright encounters an unannotated function parameter with a default value, it infers the type of the parameter based on the type of the default value. This heuristic tends to work pretty well, but it fails in some circumstances such as with the `dash` library, which uses a singleton object — an instance of the class `_UNDEFINED` — as a default value for many parameters. This results in false positive errors if someone enables type checking for this code.

I think we have several options here:
1. Leave the current behavior in place and continue to encourage library authors to add annotations or provide stubs.
2. Modify the current behavior in one of the following ways
a. If the evaluated type of the default value is an instance of a private class (one whose name starts with an underscore), assume it might be using a singleton and don't infer the type of the parameter. 
b. When inferring the type of an unannotated parameter, always include in its inferred type an `Unknown` so it accepts any argument without error. This would allow us to still present reasonable completion suggestions but avoid any false positives. On the other hand, it would likely result in false negatives.
c. When inferring the type of an unannotated parameter, limit inference to one of a handful of built-in types: `int`, `float`, `bool`, `complex`, `str`, `bytes`.

More discussion is needed to choose from among the above options.

Here's a code snippet that demonstrates the problem:
```python
from dash import Dash, html, dcc

app = Dash(__name__)
app.layout = html.Div(
    children=[
        dcc.Graph(
            className="h-screen"
        )
    ]
)
```

**Comment by spookyuser** [comment]
3. https://github.com/Microsoft/TypeScript/issues/19573#issuecomment-447826636 :)

**Comment by erictraut** [comment]
Erik De Bonte suggested another option:
d. If the evaluated type of the default value is an instance of a private class, always include in its inferred type an `Unknown`

I think this represents a good tradeoff between "a" and the current behavior with `None` default values.

**Comment by erictraut** [comment]
I've implemented option d. This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.275, which I just published. It will also be included in a future release of pylance.
